### ISSUE TITLE
emulate kana's textobj-entire

### ISSUE THREAD

**Issue by ninrod** [noctuid/targets.el#17]
@noctuid, I have this line on my emacs [config](https://github.com/ninrod/dotfiles/blob/a157b0d3e9543246b53c39830f45ceb7b3b27071/emacs/boot.org#bootstrap):

```elisp
;; simulation of kana's textobj-entire
(evil-define-text-object ninrod/textobj-entire (count &optional beg end type)
    (evil-range (point-min) (point-max)))
(define-key evil-outer-text-objects-map "e" 'ninrod/textobj-entire))
```

And with just that, I'm already able to issue, for example, `yae` to copy the entire buffer into the kill-ring, [kana's textobject-entire](https://github.com/kana/vim-textobj-entire) style.

In preparation for my work in #4, and as a follow-up to your [comment](https://github.com/noctuid/targets.el/issues/14#issuecomment-274918557) in #14, and considering [this](https://github.com/noctuid/targets.el#miscellaneoussimple) section of the readme, maybe I could implement this `e` text object first to get a hang of how to correctly implement things in `targets.el`, since it appears to be simpler than the `xml` text object in #4.

So I'm thinking that I first I should add something to `targets-object-text-objects`, but I don't exactly know what that does. For instance, `'evil-paragraph` does not appear to be a function nor a variable. I tried to find it through `C-h f` and `C-h v`.

Like #4, should I be required to write a seeking function for that as well? At first, this does not appear to make sense because there's no need to seek forward or backward to the entire buffer. point is always in entire-buffer territory no matter where point is. So maybe this is a special case?

**Comment by noctuid** [comment]
Yes, this is a special case. You wouldn't need to touch `targets-object-text-objects`; this text object should be added separately from the others under the `Specific Text Objects` heading (since seeking and next/last text objects don't make sense). All that would need to be done for this is to create two text objects with `targets--define-text-object` (e.g. `(targets--define-text-object targets-a-buffer "Select a buffer." nil (evil-range ...))`). The inner text object would just exclude whitespace (e.g. using `skip-chars-(forward|backward)`).

**Comment by ninrod** [comment]
hi @noctuid, just letting you know that this is surely on my radar. I've just had been really busy, no time to hack. But I absolutely love this package and appreciate you work on this. As soon as I get some time on my hands I plan to dig deeper and write some targets.el code.
